app/javascript/lib/scp-api.js

```diff
@@ -257,6 +304,27 @@ export async function deleteStudyFile(studyAccession, fileId, mock=false) {
 }
 
 
+/**
+ * Fetches a given resource from a GCP bucket -- this handles adding the
+ * appropriate SCP readonly bearer token, and using the Google API URL that allows CORS
+ *
+ * @param {String} bucketName bucket name
+ * @param {String} fileName file name
+*/
+export async function fetchBucketFile(bucketName, fileName, mock=false) {
+  const init = {
+    method: 'GET',
+    headers: {
+      Authorization: `Bearer ${window.SCP.readOnlyToken}`
```

Comment:
Yeah, we will have to put in extra error handling anyway for the ToS refactoring & PPW, so it might be expedient to start building a framework for doing that now.  The likelyhood of getting `401` or `403` when trying to generate tokens will definitely go up for private studies (public will be unaffected).